### ISSUE TITLE
Research (and fix) state not found warning

### ISSUE THREAD

**Issue by oxarbitrage** [ZcashFoundation/zebra#8876]
We are recently seeing a warning in zebrad. We want to check it in more details and decide what to in the context of this ticket.

https://github.com/ZcashFoundation/zebra/pull/8865#issuecomment-2352971509

```
2024-09-16T13:37:54.535314Z  WARN zebra_state::service::finalized_state::disk_db: could not canonicalize "/var/cache/zebrad-cache/state/v25/mainnet": No such file or directory (os error 2)
```

In regtest mode:
```
2024-09-19T11:59:37.538954Z  WARN zebra_state::service::finalized_state::disk_db: could not canonicalize "/tmp/testp3siw8dm/515/node0/state/v25/regtest": No such file or directory (os error 2)
```

**Comment by mpguerra** [comment]
Is this only in CI?

**Comment by arya2** [comment]
> Is this only in CI?

No, I've seen it locally as well. I think it happens when creating a new empty state, or after the old major version has already been renamed and Zebra has restarted because there's no file there. We probably want it to be an info-level log instead, or check for a version file in the new major version before attempting to canonicalize the path to the old major version.

**Comment by arya2** [comment]
@natalieesk this is a very small issue, we just need to change a warning-level log to info-level.
